router/router.go

```diff
@@ -169,6 +173,8 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 	}()
 
+	req = req.WithContext(progrock.RecorderToContext(req.Context(), r.recorder))
```

Comment:
This shimmies the recorder through `context.Context` so `ToResolver` can use it.